src/cdn_definitions/data.yaml

```diff
@@ -92,3 +92,136 @@ origin_alias:
 # permanent.
 - src: /origin/rpm
   dest: /origin/rpms
+
```

Comment:
This seems to be a snapshot of real data, maybe cut down a little, but we know the real data is meant to come from cdn-definitions-private.

Is the idea then that the reference data is initially populated from a snapshot but then allowed to diverge over time?

I don't think it's a good idea to set it up that way for a couple of reasons:

- If this data looks like real data then people are going to wrongly assume it's supposed to be updated whenever changes to the real data are made.  We've already seen exactly this problem in cdn-utils repo, e.g. see https://code.engineering.redhat.com/gerrit/#/c/198307/ where, as well as bumping the real version numbers used from within code, someone has wrongly thought it's necessary to update the testdata as well - because it looks the same.  I think that people just grep for the magic numbers/versions they know need to be updated and don't necessarily look at the docs first (or they assume the docs aren't complete).
- If the data is realistic then there is a risk that someone can develop code against it and not even realize the existence of the real data in a separate repo until problems occur.  e.g. imagine that someone writes some code which looks at the symlink_exceptions.  Then imagine we need to define a new exception, and we do that in cdn-definitions-private and adjust CDN accordingly.  Some time later this person files a bug against us stating "Hey, I've noticed this path doesn't use a symlink but you didn't list it in the exceptions!"  Then we have to explain that they should have deployed their tool/service to load from cdn-definitions-private if they wanted production data.  To be honest, I would even worry that whatever ticket master saw that complaint first is not going to realize they should tell the reporter to look at cdn-definitions-private, and they might instead waste time updating the field here similarly as in the gerrit change above.

In summary I worry that this data looking realistic is going to result in long-term confusion with people wrongly assuming that the realistic-looking data is *real* data.

Could we go through and force it to more obviously look like bogus/dummy data?  The idea would be that just about anyone looking at this file could quickly come to the conclusion "alright, this clearly isn't the true configuration used for production CDN right now, I need to go find where that's stored".

I get the impression this idea is already being applied somewhat as I note some differences here, e.g. the real RHUI ID isn't 222.  Can we push it further?

Reply:
+1. I think the data appears less realistic now.